### PR TITLE
Add ReportingObserver API.

### PR DESCRIPTION
**Pull request by jpmedley**
On workers: https://www.chromestatus.com/feature/6310316751192064
Initial support: https://www.chromestatus.com/feature/4691191559880704
What I can determine of other browsers: https://web-confluence.appspot.com/#!/catalog?releases=%5B%22Edge_18.17763_Windows_10.0%22,%22Safari_13.0.2_OSX_10.15%22,%22Chrome_78.0.3904.70_Windows_10.0%22,%22Firefox_70.0_Windows_10.0%22%5D&q=%22ReportingOb%22

### CODE REVIEW

File: api/ReportingObserver.json [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            "version_added": false
+          },
+          "firefox_android": {
+            "version_added": false
+          },
+          "ie": {
+            "version_added": false
+          },
+          "opera": {
+            "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:26]:
```suggestion
            "version_added": "56"
```

File: api/ReportingObserver.json [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+          },
+          "opera_android": {
+            "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:29]:
```suggestion
            "version_added": "48"
```

File: api/ReportingObserver.json [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+          },
+          "safari": {
+            "version_added": false
+          },
+          "safari_ios": {
+            "version_added": false
+          },
+          "samsunginternet_android": {
+            "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:38]:
```suggestion
            "version_added": "10.0"
```

File: api/ReportingObserver.json [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+              "version_added": false
+            },
+            "firefox_android": {
+              "version_added": false
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:74]:
```suggestion
              "version_added": "56"
```

File: api/ReportingObserver.json [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:77]:
```suggestion
              "version_added": "48"
```

File: api/ReportingObserver.json [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:86]:
```suggestion
              "version_added": "10.0"
```

File: api/ReportingObserver.json [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+              "version_added": false
+            },
+            "firefox_android": {
+              "version_added": false
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:122]:
```suggestion
              "version_added": "56"
```

File: api/ReportingObserver.json [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:125]:
```suggestion
              "version_added": "48"
```

File: api/ReportingObserver.json [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:134]:
```suggestion
              "version_added": "10.0"
```

File: api/ReportingObserver.json [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+              "version_added": false
+            },
+            "firefox_android": {
+              "version_added": false
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:170]:
```suggestion
              "version_added": "56"
```

File: api/ReportingObserver.json [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:173]:
```suggestion
              "version_added": "48"
```

File: api/ReportingObserver.json [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:182]:
```suggestion
              "version_added": "10.0"
```

File: api/ReportingObserver.json [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+              "version_added": false
+            },
+            "firefox_android": {
+              "version_added": false
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:218]:
```suggestion
              "version_added": "56"
```

File: api/ReportingObserver.json [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:221]:
```suggestion
              "version_added": "48"
```

File: api/ReportingObserver.json [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:230]:
```suggestion
              "version_added": "10.0"
```

File: api/ReportingObserver.json [hunk 16]
```diff
@@ -0,0 +1,296 @@
...
+              "version_added": false
+            },
+            "ie": {
+              "version_added": false
+            },
+            "nodejs": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:269]:
```suggestion
              "version_added": "70"
```

File: api/ReportingObserver.json [hunk 17]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "opera_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:272]:
```suggestion
              "version_added": false
```

File: api/ReportingObserver.json [hunk 18]
```diff
@@ -0,0 +1,296 @@
...
+            },
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "samsunginternet_android": {
+              "version_added": true
```
Comment (ddbeck) [inline api/ReportingObserver.json:281]:
```suggestion
              "version_added": false
```

**Comment by jpmedley** [comment]
done. Please take another look.

**Comment by ddbeck** [comment]
There's a few changes you didn't apply. Did I get something wrong there? If they don't apply, that's fine, I'd just like to have a record of why. 😄

**Comment by jpmedley** [comment]
@ddbeck You didn't miss anything. I missed some, probably because of the stupid way scroll position jumps around when you commit suggestions. I think I got them all.

**Pull request merged**